Automattic/wp-calypso
# Jetpack Connection: fix sign-up as a different user

## Proposed Changes
During Jetpack connection flow, logged-in WPCOM users see the "Sign in as a different user" button, where users can enter a username or an email
If the entered email is not registered and you intend to register, you get redirected to the "My Home" page of your default site.

This happens because the form emails the user a "magic link" and redirects them to the "Magic link send" page. However, that page cannot be displayed as the user is still logged in under the old WPCOM account, so they get redirected to the homepage instead.

This PR will logout users as soon as they click "Sign in as a different user" button, so their other WPCOM account won't interfere.

1191179647901802-as-1203666065057524
p8oabR-11U-p2#comment-6947

## Testing Instructions
_Disable all cookie protection in browser settings, otherwise the flow will not work due to test/live domain discrepancies in API requests._

1. Open the "Calypso Live" link below in a private window, login.
2. Log into WP.com under the same user.
3. Open a Jetpack site, Initiate WP connection flow.
4. When you see the "Approve" page, replace `wordpress.com` with the `*.calypso.live` URL that got generated by "Calypso Live" link. The page should look the same.
5. Click "Sign in as a different user". You should see the field to enter an email.
6. Switch the domain back to `https://wordpress.com/` because the "Calypso Live" doesn't seem to have magic link emails enabled. There are no changes on that page.
7. Enter a registered (but different) email/username, confirm you got logged in and see the "Approve" page again.
8. Repeat steps 1-5, and enter a non-registered email.
9. You should see the "Magic link sent" page, and actually receive the email.
10. Due to live/test domain name discrepancy, the magic link from email will not send you back to "Approve" page (that'll be tested after merging the PR), but it should create the WPCOM user account.

_Don't forget to enable back the cookie protection settings!_

## Pre-merge Checklist
